packages/@aws-cdk/aws-stepfunctions/lib/states/task-base.ts

```diff
@@ -0,0 +1,310 @@
+
+import * as cdk from '@aws-cdk/core';
+
+/**
+ * Props that are common to all tasks
+ */
+export interface TaskStateBaseProps {
+  /**
+   * An optional description for this state
+   *
+   * @default - No comment
+   */
+  readonly comment?: string;
+
+  /**
+   * JSONPath expression to select part of the state to be the input to this state.
+   *
+   * May also be the special value DISCARD, which will cause the effective
+   * input to be the empty object {}.
+   *
+   * @default - The entire task input (JSON path '$')
+   */
+  readonly inputPath?: string;
+
+  /**
+   * JSONPath expression to select select a portion of the state output to pass
+   * to the next state.
+   *
+   * May also be the special value DISCARD, which will cause the effective
+   * output to be the empty object {}.
+   *
+   * @default - The entire JSON node determined by the state input, the task result,
+   *   and resultPath is passed to the next state (JSON path '$')
+   */
+  readonly outputPath?: string;
+
+  /**
+   * JSONPath expression to indicate where to inject the state's output
+   *
+   * May also be the special value DISCARD, which will cause the state's
+   * input to become its output.
+   *
+   * @default - Replaces the entire input with the result (JSON path '$')
+   */
+  readonly resultPath?: string;
+
+  /**
+   * Timeout for the state machine
+   *
+   * @default - None
+   */
+  readonly timeout?: cdk.Duration;
+
+  /**
+   * Timeout for the heartbeat
+   *
+   * @default - None
+   */
+  readonly heartbeat?: cdk.Duration;
+
+  /**
+   * AWS Step Functions integrates with services directly in the Amazon States Language.
+   * You can control these AWS services using service integration patterns
+   *
+   * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html#connect-wait-token
+   *
+   * @default IntegrationPattern.REQUEST_RESPONSE
+   */
+  readonly integrationPattern?: IntegrationPattern;
+}
+
+/**
+ * Define a Task state in the state machine
+ *
+ * Reaching a Task state causes some work to be executed, represented by the
+ * Task's resource property. Task constructs represent a generic Amazon
+ * States Language Task.
+ *
+ * For some resource types, more specific subclasses of Task may be available
+ * which are more convenient to use.
+ */
+export abstract class TaskStateBase extends State implements INextable {
+
+  public readonly endStates: INextable[];
+
+  protected abstract readonly taskMetrics: TaskMetricsConfig | undefined;
+  protected abstract readonly taskPolicies: iam.PolicyStatement[] | undefined;
+
+  constructor(scope: cdk.Construct, id: string, props: TaskStateBaseProps) {
+    super(scope, id, props);
+    this.endStates = [this];
+  }
+
+  /**
+   * Add retry configuration for this state
+   *
+   * This controls if and how the execution will be retried if a particular
+   * error occurs.
+   */
+  public addRetry(props: RetryProps = {}): TaskStateBase {
+    super._addRetry(props);
```

Comment:
It does. however, the way we've modeled [state](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-stepfunctions/lib/states/state.ts#L6-L8) (which the base class extends) is to be a union of all properties shared by all states and we have the extending classes expose the relevant bits. Since this class represents a task state, it made sense to reuse retry and catch (as the current `Task` and `Parallel` states do). 

I'm not opposed to the idea of a clearer division of responsibility per state. Since we could have users already directly extending the State class and using it's properties, this functionality will still have to remain there in the superclass (`State`).